tests/index.test.ts

```diff
@@ -0,0 +1,89 @@
+import { ExtensibilitySdk, Message, OAuthDialogCompletedMessage } from '../src/index';
+import runtime from '../src/sdk/RuntimeContext';
+import oauthService from '../src/sdk/services/oauthService';
+
+describe('sdk tests', () => {
+  let sdk: ExtensibilitySdk;
+
+  describe.only('receiving messages', () => {
+    var messageHandler: EventListenerOrEventListenerObject;
+    let originalEventListener: (event: string, handler: EventListenerOrEventListenerObject) => void;
+    let originalOpenPopup: (redirectUri?: string, state?: { [key: string]: string }) => void;
+
+    beforeEach(async () => {
+      window.setTimeout = jest.fn().mockImplementation(() => 123) as any;
+
+      originalOpenPopup = oauthService.openPopup;
+      oauthService.openPopup = jest.fn();
+
+      originalEventListener = window.addEventListener;
+      window.addEventListener = (event: string, handler: EventListenerOrEventListenerObject): void => {
```

Comment:
you are overwriting the `addEventListener` for all events but not processing events other than `message`

Reply:
Our SDK is using only message event listener - added throwing of an error if event name is not a "message"